main.go

```
@@ -339,17 +341,11 @@ func (d *jfsDriver) Remove(r *volume.RemoveRequest) error {
 		return logError("volume %s is in use", r.Name)
 	}
 
-	if err := umountVolume(v); err != nil {
-		return err
-	}
-
-	if err := os.Remove(v.Mountpoint); err != nil {
-		return logError(err.Error())
-	}
+	// umount, ignore error
+	umountVolume(v)
```

这里的 remove 可以保留，先判断挂载点是否还存在，不存在再 remove

---

`umountVolume` 里有加了 `remove` 逻辑

---

Unmount 的时候已经执行过卸载和 remove，正常情况下，这里重复执行会报错

---

> Unmount 的时候已经执行过卸载和 remove，正常情况下，这里重复执行会报错

`// umount, ignore error`
注释里写了，这里有故意地忽略错误

---

重复执行的目的是什么呢？兜底的话，应该先检查挂载点是否存在，不存在再执行 umount + remove，此时执行失败需要报错

---

这里保留，是保留之前的逻辑；
如果确定任何情况（强制删除？）下都不会有跳过 Unmount 直接 Remove 的情况，可以删掉

---

建议先检查挂载点是否存在，不存在再执行 umount + remove，此时执行失败需要报错

---

> 重复执行的目的是什么呢？兜底的话，应该先检查挂载点是否存在，不存在再执行 umount + remove，此时执行失败需要报错

不存在...执行 umount 干啥...
检查挂载点存不存在，意义不大；对本插件有实际意义的就是挂载情况下的 umount + remove，其他意义不大

---

打错字了，存在再执行 umount + remove

---

Done